python/cudf/cudf/utils/hash_vocab_utils.py

```diff
@@ -130,13 +131,13 @@ def _perfect_hash(integers, max_constant):
         bin_length = len(internal_table)
         max_bin_length = max(bin_length, max_bin_length)
         internal_table_coeffs[i] = (
-            coeff_a << A_SECOND_LEVEL_SHIFT_AMT
-            | coeff_b << B_SECOND_LEVEL_SHIFT_AMT
-            | bin_length
-        )
-        offset_into_flattened_table[i + 1] = (
-            offset_into_flattened_table[i] + bin_length
+            np.uint64(coeff_a) << A_SECOND_LEVEL_SHIFT_AMT
```

Comment:
*question*: Why we we need to cast these to `uint64`?

Reply:
Opposite of the NumPy 2 above, NumPy 1 uses the logic: `np.uint64(1) + 1` being the same as `np.uint64(1) + np.int64(1)` since both are scalars.  But *grrrrrrr*, that promotes to float64 (which luckily doesn't work at all for bit-shifts)...

Reply:
Ah yeah, this one I hate :P, thanks!